test/recipes/04-test_encoder_decoder_legacy.t

```diff
@@ -22,6 +22,6 @@ plan tests => 1;
 
 
 $ENV{OPENSSL_MODULES} = abs_path(bldtop_dir("providers"));
-$ENV{OPENSSL_CONF} = abs_path(srctop_file("test", "default-and-legacy.cnf"));
+$ENV{OPENSSL_CONF} = abs_path(srctop_file("test", "default.cnf"));
```

Comment:
Shouldn't this be:

``` perl
my $conf = disabled('legacy') ? 'default.cnf' : 'default-and-legacy.cnf';
$ENV{OPENSSL_CONF} = abs_path(srctop_file('test', $conf));
```

Reply:
... and likewise in other places...

Reply:
No?

AFAICT, this test has nothing to do with the legacy provider at all. The name of the test is somewhat misleading. It is not about legacy *algorithms* but about legacy *APIs*. In terms of algorithms it uses DH, DSA, EC, ED25519, ED448, X25519, X448, RSA, and RSA-PSS. None of these are in the legacy provider, so it is unnecessary to load it.

Reply:
Hmmmm, looks like you're right.  I could've sworn I was getting in trouble without the legacy provider at some point, but that may have been while still experimenting a lot, and a later cleanup removed that necessity.

Ok then